presto-main/src/main/java/com/facebook/presto/cost/PlanNodeCostEstimate.java

```diff
@@ -108,20 +165,27 @@ public boolean equals(Object o)
             return false;
         }
         PlanNodeCostEstimate that = (PlanNodeCostEstimate) o;
-        return Double.compare(that.cpuCost, cpuCost) == 0 &&
+        return Double.compare(that.individualCpuCost, individualCpuCost) == 0 &&
+                Double.compare(that.individualMemoryCost, individualMemoryCost) == 0 &&
+                Double.compare(that.individualNetworkCost, individualNetworkCost) == 0 &&
+                Double.compare(that.cpuCost, cpuCost) == 0 &&
                 Double.compare(that.memoryCost, memoryCost) == 0 &&
                 Double.compare(that.networkCost, networkCost) == 0;
     }
 
     @Override
     public int hashCode()
     {
-        return Objects.hash(cpuCost, memoryCost, networkCost);
+        return Objects.hash(individualCpuCost, individualMemoryCost, individualNetworkCost,
+                cpuCost, memoryCost, networkCost);
     }
 
     public PlanNodeCostEstimate add(PlanNodeCostEstimate other)
     {
         return new PlanNodeCostEstimate(
+                individualCpuCost,
+                individualMemoryCost,
+                individualNetworkCost,
```

Comment:
why not the `other. individualNetworkCost`? `Add` until this point was commutative. Maybe `Double.NaN` would be better here.